### PR TITLE
Mrinmoy

### PR DESCRIPTION
**Pull request by GithubMrinmoy**
@bdrekapil @mishisinku please review.

### CODE REVIEW

File: md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java [hunk 1]
```diff
@@ -20,12 +20,13 @@
...
 import java.io.PrintWriter;
 import java.util.ArrayList;
+import java.util.List;
 
 /**
  * Created by arijit on 12/8/14.
  */
 public class BaseStructure {
-    private static Logger LOGGER = Logger.getLogger(BaseStructure.class);
+    private static final Logger LOGGER = Logger.getLogger(BaseStructure.class);
```
Comment (bdre-bda) [inline md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java:29]: MAJOR Remove this unused "LOGGER" private field. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1068)

File: md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java [hunk 2]
```diff
@@ -48,6 +39,27 @@ public void setBusDomainId(Integer busDomainId) {
     private String nextProcessIds;
     private String batchCutPattern;
     private Boolean deleteFlag;
+    List<ProcessInfo> processInfo;
```
Comment (bdre-bda) [inline md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java:42]: MAJOR Rename field "processInfo" [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1700)

File: md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java [hunk 3]
```diff
@@ -173,8 +175,32 @@ public void persistBeanList(List beans, boolean printOnly) {
             } catch (IllegalAccessException e) {
                 LOGGER.error(e);
                 throw new MetadataException(e);
-            }
+            }*/
+            persistBeanTryCatch(bean);
         }
+        /*LOGGER.info("propertiesToBeSaved=" + propertiesToBeSaved);
```
Comment (bdre-bda) [inline md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java:181]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java [hunk 4]
```diff
@@ -162,7 +164,7 @@ public void persistBeanData(Object bean, boolean printOnly) {
 
     public void persistBeanList(List beans, boolean printOnly) {
         for (Object bean : beans) {
-            try {
+           /* try {
```
Comment (bdre-bda) [inline md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java:167]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java [hunk 5]
```diff
@@ -135,7 +135,7 @@ private void persistSingleBean(Object bean) throws IntrospectionException, Invoc
      */
     public void persistBeanData(Object bean, boolean printOnly) {
 
-        try {
+        /*try {
```
Comment (bdre-bda) [inline md-commons/src/main/java/com/wipro/ats/bdre/util/OozieUtil.java:138]: MAJOR This block of commented-out lines of code should be removed. [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by bdre-bda** [comment]
SonarQube analysis reported 20 issues:
- BLOCKER 2 blocker
- CRITICAL 2 critical
- MAJOR 7 major
- MINOR 9 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
- MINOR [The Cyclomatic Complexity of this method "populateBeans" is 16 which is greater than 10 authorized.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/data-lineage/src/main/java/com/wipro/ats/bdre/lineage/api/PersistenceUnit.java#L85) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
- MINOR [Replace all tab characters in this file by sequences of white-spaces.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/data-lineage/src/main/java/com/wipro/ats/bdre/lineage/api/PersistenceUnit.java) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00105)
- CRITICAL [Remove this call to "exit" or ensure it is really required.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java#L94) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1147)
- MAJOR [Replace this usage of System.out or System.err by a logger.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java#L93) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS106)
- CRITICAL [Remove this hard-coded password.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/DQSetupInfo.java#L33) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS2068)
- MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java#L96) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1213)
- MAJOR [Method has 9 parameters, which is greater than 7 authorized.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java#L117) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS00107)
- MINOR [Move this constructor to comply with Java Code Conventions.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java#L101) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1213)
- MINOR [Move this constructor to comply with Java Code Conventions.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java#L117) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1213)
- MINOR [Move this variable to comply with Java Code Conventions.](https://github.com/WiproOpenSourcePractice/openbdre/blob/f7bd8ab38b9e8198c1dfcd6e4516bab6cab9e02a/md-commons/src/main/java/com/wipro/ats/bdre/md/beans/table/ArchiveConsumpQueue.java#L117) [rule](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1213)
- ... 5 more

**Comment by mishisinku** [comment]
@GithubMrinmoy  Hi Mrninmoy. Please create a subtask in jira under main task 111 and then make the corrections . Next make sure the commit message should have meaningful sentence along with the subtask jira no. e.g " Fixing code quality issues in md commons BDRE-188".

File: md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java (hunk 2 above)
Comment (GithubMrinmoy) [inline md-commons/src/main/java/com/wipro/ats/bdre/md/beans/ProcessInfo.java:42]: @bdrekapil please suggest correction here.

**Comment by GithubMrinmoy** [comment]
@mishisinku  Hi Mishi. I have used "BDRE-169 Sonar code quality correction in md-commons" as commit message and BDRE-169 is subtask. Is there anything to change?

File: md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java (hunk 1 above)
Comment (GithubMrinmoy) [inline md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java:29]: @bdrekapil should I remove this LOGGER line?

Comment (bdrekapil) [inline md-commons/src/main/java/com/wipro/ats/bdre/BaseStructure.java:29]: Either remove it or use it to provide some information or debug.

**Comment by wiprodev** [comment]
@GithubMrinmoy it's a rule to include one or more Jira number in both commit message or pull request message. Pull request should include a few major Jira ticket number from the commits.

**Pull request closed without merging**